### ISSUE TITLE
Training issues: Memory and training details

### ISSUE THREAD

**Issue by Ivan-Fan** [royorel/Lifespan_Age_Transformation_Synthesis#32]
Hi! This is truly splendid work! 

1) The first question is about the CUDA memory. Right now I have downloaded all the ffhq-aging dataset to my gdrive and try to train on colab (Tesla T4 was allocated to me this time, with 15GB memory). It turned out that only a batch size of 2 consumes 11GB memory on this graphic card and 3 will alert OOM. I was wondering is there any way to increase the batch size and speed up the training a little bit? E.g., modify some arguments in the options?

I noticed that it is said in the paper "Each model was trained with a batch size of 12 for 400 epochs on 4 GeForce RTX 2080 Ti GPUs." So I assume it is possible to train with at least a batch size of 3 on this card? 

2) The second question is, during which epoch of training should I expect to see a human-face-like result? Maybe 100 epochs? As I am training very slowly, I am not sure when I will see some promising results :( There is nearly nothing in the first 2 epochs.

Sorry for my stupid question :) I am a new learner but really enjoy this kind of topic.

**Comment by royorel** [comment]
Hi @Ivan-Fan,

The script was not built for training on Colab, so I can't really comment on what's going on and where the overhead is coming from.

Technically speaking, you can tweak the code to accumulate gradients in the forward pass (e.g. run a batch size of say, 1 but taking an optimizer step every 3 forward passes). That will allow you to reduce the batch size an fit to the GPU memory while essentially maintaining the same functionality.

Regarding epochs, the training can be unstable sometime. I would say that if you don't see faces a bit after the first learning rate decay you should restart the process. You can also lower the initial learning rate, but that comes at the expense of the quality of the final results.

**Comment by Ivan-Fan** [comment]
Really thanks! I will try!

**Comment by Ivan-Fan** [comment]
Hi, sorry to bother again :( I met an error showing like this after I tried to load a pretrained model on colab:
`python train.py --sort_order '0-2,7-9' --dataroot 'datasets/males' --batchSize 2 --print_freq 50 --display_id 0 --name 'yager0307_young' --save_epoch_freq 1 --which_epoch 53 --continue_train True`:
![error1](https://user-images.githubusercontent.com/39240631/157280716-f722501d-5fb1-430b-9fac-b239573c86ec.png)
![error2](https://user-images.githubusercontent.com/39240631/157280735-44996344-adf6-41ee-8a4f-2b4855a5d1f1.png)

Is it something to do with colab?

**Comment by royorel** [comment]
@Ivan-Fan
 
It loos like you didn't open the visdom server.

You need to run it in the background. `python -m visdom.server`

For further instructions you can go to:
https://github.com/fossasia/visdom

**Comment by Ivan-Fan** [comment]
Thanks! It really helps! @royorel
